### PR TITLE
[PLATFORM-3258] Update CLI to algo rust version 3 crate. Keep OpenSSL.

### PR DESCRIPTION
**Pull request by bytekast**

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -30,16 +30,16 @@ pub struct Profile {
...
+            git_server: git_server.map(|s| s.as_str().trim_end_matches('/').to_owned()),
             api_key: api_key,
         }
     }
 
     pub fn client(&self) -> Algorithmia {
         match self.api_server {
-            Some(ref api) => Algorithmia::client_with_url(api, &*self.api_key),
-            None => Algorithmia::client(&*self.api_key),
+            Some(ref api) => Algorithmia::client_with_url(api, &*self.api_key).unwrap(),
```
Comment (anowell) [inline src/config.rs]: `client_with_url` now takes the api_key first (super confusing breaking change, but more consistent with other clients)

Comment (anowell) [reply src/config.rs]: https://docs.rs/algorithmia/3.0.0-beta/algorithmia/struct.Algorithmia.html#method.client_with_url

**Review by anowell — COMMENTED** [review]
Everything else here so far looks good.

File: src/config.rs (hunk 1 above)
Comment (bytekast) [reply src/config.rs]: gotcha. thanks!

**Comment by bytekast** [comment]
Closing this in favor of https://github.com/algorithmiaio/algorithmia-cli/pulls/26

**Pull request closed without merging**